api/objects/validate

```diff
@@ -79,18 +79,28 @@ case "update-service":
     }
 
     if($data['Ports']) {
-        if (strpos($data['Ports'][0], "-") !== false ) { # port range
-            $tmp = explode("-",$data['Ports'][0]);
-            # two non-zero integers, right number must be greater then left one
-            if ( !isset($tmp[0], $tmp[1]) || !(int)$tmp[0] || !(int)$tmp[1] || (int)$tmp[0] >= (int)$tmp[1] ) {
-                $v->addValidationError('Ports', 'not_valid_port_range');
+        $portValidator = $v->createValidator(Validate::PORTNUMBER);
+        # Verify that we use either a list OR a range of ports
+        if (count($data['Ports']) == 1 and strpos($data['Ports'][0], "-") !== false ) { # port range first element of array
+            list($lower_port, $upper_port) = explode("-", $data['Ports'][0], 2);
+            # Verify the two sides are port values
+            if (!$portValidator->evaluate($lower_port) || !$portValidator->evaluate($upper_port)) {
+                $v->addValidationError('Ports', 'not_valid_port_range', $data['Ports'][0]);
+            }
+            # Verify the right number is greater than left one
+            if ( $lower_port >= $upper_port ) {
```

Comment:
If $upper_port is 0 both conditions are met. Is it handled by the UI? Maybe it's better to limit to one error per field?

```suggestion
            elseif ( $lower_port >= $upper_port ) {
```